src/nix/shell.cc

```diff
@@ -245,7 +245,7 @@ struct CmdDevShell : Common
 
         stopProgressBar();
 
-        auto shell = getEnv("SHELL", "bash");
+        auto shell = "bash";
```

Comment:
This doesn't compile because it infers the wrong type for `shell` (`char *` rather than `std::string`).